### ISSUE TITLE
how to remove an entry from history

### ISSUE THREAD

**Issue by jerabaul29** [xonsh/xonsh#4359]
How to remove an entry from history with a simple command?

In one of my Xonsh sessions, I inadvertently entered an important password. I can see it (and the history entry number) using:

```
history -n
```

However, trying to remove it with the classical:

```
history -d ENTRY_NUMBER
```

Gives me:

```
$ history -d 15                                                                           
history: error: cannot convert '-d' to slice
```

**Comment by jnoortheen** [comment]
the history command doesn't have a way to remove an entry. Do you want to work on it?

**Comment by jerabaul29** [comment]
Ok, so I understand that the history command is implemented fully in Xonsh then, right? :) I could do that, sounds interesting :)

**Comment by jerabaul29** [comment]
I would need to implement it here, right? :)

**Comment by jnoortheen** [comment]
Yes it has sqlite and json backends. You need to implement for them as well.

**Comment by gforsyth** [comment]
@jerabaul29 -- it would be awesome to have something in the `history` command to remove individual commands.
If you need to _quickly_ remove that particular command, assuming you are using the (default) `json` backend, you can use `ack` or `ag` or `grep` to find the particular `json` file that has the offending entry in it and either nuke the whole file or open it up and strip out the individual json entry that has your password in it.
